internal/reminder/reminder.go

```diff
@@ -144,19 +147,64 @@ func (r *reminder) sendReminders(ctx context.Context) []error {
 
 	logger.Info().Msgf("created repository batch of size: %d", len(repos))
 
-	// Update the reminder_last_sent for each repository to export as metrics
+	messages := make([]*message.Message, 0, len(repos))
+	errorSlice := make([]error, 0)
+
+	tx, err := r.store.BeginTransaction()
+	if err != nil {
+		return []error{err}
+	}
+
+	defer tx.Rollback()
+
+	qtx := r.store.GetQuerierWithTransaction(tx)
+
+	// TODO: Collect Metrics
+	// Potential metrics:
+	// - Gauge: Number of reminders in the current batch
+	// - UpDownCounter: Average reminders sent per batch
+	// - Histogram: reminder_last_sent time distribution
 	for _, repo := range repos {
-		logger.Debug().Str("repo", repo.ID.String()).
-			Time("previously", repo.ReminderLastSent.Time).Msg("updating reminder_last_sent")
-		err := r.store.UpdateReminderLastSentById(ctx, repo.ID)
+		repoReconcilerMessage, err := remindermessages.NewRepoReminderMessage(
+			repo.ProviderID, repo.RepoID, repo.ProjectID,
+		)
+		if err != nil {
+			errorSlice = append(errorSlice, err)
+			continue
+		}
+
+		logger.Debug().
+			Str("repo", repo.ID.String()).
+			Time("previously", repo.ReminderLastSent.Time).
+			Msg("updating reminder_last_sent")
+
+		err = qtx.UpdateReminderLastSentById(ctx, repo.ID)
 		if err != nil {
 			logger.Error().Err(err).Str("repo", repo.ID.String()).Msg("unable to update reminder_last_sent")
-			return []error{err}
+			errorSlice = append(errorSlice, err)
+			continue
+		}
+
+		messages = append(messages, repoReconcilerMessage)
+	}
+
+	if len(messages) != 0 {
+		err = r.eventPublisher.Publish(events.TopicQueueRepoReminder, messages...)
+		if err != nil {
+			errorSlice = append(errorSlice, fmt.Errorf("error publishing messages: %w", err))
+		} else {
```

Comment:
Rather than an `if/else` here, generally prefer to use early-return, particularly since this seems like a terminal case.

More generally, I'm not convinced that there's a large benefit to batching the Publish messages, rather than sending each in the previous part of the loop.

Reply:
Done.

> I'm not convinced that there's a large benefit to batching the Publish messages, rather than sending each in the previous part of the loop.

Bulk updates might be better than connecting to the db for every update, I went with bulk updates with slight changes in logic.